File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/lab/BinaryPlus.java
// BinaryPlus.java

import java.io.PrintWriter;

public class BinaryPlus extends Binary
{
	public BinaryPlus(Expression aOperand1, Expression aOperand2)
	{
		super(aOperand1, aOperand2);
	}

	public void output(PrintWriter aPrintWriter)
	{
		outputInfix(aPrintWriter, "+");
	}
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/interpreter/lab/binary_plus.cpp
// binary_plus.cpp

#include <iostream>
#include "binary.h"
#include "binary_plus.h"
#include "dp_exception.h"
class expression;

binary_plus::binary_plus(expression * a_operand1, expression * a_operand2)
: binary(a_operand1, a_operand2)
{
}

void binary_plus::output(std::ostream & a_ostream) const
{
	output_infix(a_ostream, "+");
}
